src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -2863,6 +2863,11 @@ NeedsDistributedPlanning(Query *queryTree)
 
 	if (hasLocalRelation && hasDistributedRelation)
 	{
+		if (InsertSelectIntoLocalTable(queryTree))
+		{
+			ereport(ERROR, (errmsg("cannot plan query, Citus only supports INSERT INTO "
+								   "dist_table SELECT ... FROM local_table")));
```

Comment:
This is not quite accurate, we also support all sorts of other SELECTs (e.g. `FROM dist_table`). Maybe just say that `INSERT INTO local_table SELECT ... FROM dist_table...` is not supported?